treescript/src/treescript/github/versionmanip.py

```diff
@@ -75,15 +79,27 @@ async def bump_version(client: GithubClient, task: Dict) -> None:
     branch = get_branch(task)
     bump_info = get_version_bump_info(task)
 
-    changes = await do_bump_version(client, bump_info["files"], bump_info["next_version"], branch)
-    if changes and should_push(task, []):
-        commit_msg = "Automatic version bump CLOSED TREE NO BUG a=release"
-        if get_dontbuild(task):
-            commit_msg += DONTBUILD_MSG
+    log.info(f"Version bumping {branch} branch of {client.owner}/{client.repo} to {bump_info['next_version']}")
+
+    changes, diff = await do_bump_version(client, bump_info["files"], bump_info["next_version"], branch)
+
+    if not changes:
+        log.warn("No changes to commit!")
+        return
+
+    commit_msg = "Automatic version bump CLOSED TREE NO BUG a=release"
+    if get_dontbuild(task):
+        commit_msg += DONTBUILD_MSG
+
+    push = should_push(task, [])
+    verb = "Committing" if push else "Would commit"
```

Comment:
Question: I'm not sure why the logic changed compared the smaller exiting function. What does this new logic solve?

Reply:
Previously because we had a repo, we used to call a `log_outgoing` function that would basically just run `git log` to display the commits that are about to be pushed. But since we don't have a repo anymore, we need some logic to display what's going to be pushed in the log. I settled on building a diff and outputting that (which is what this logic is doing).

Reply:
You can see an example of this in [this task](https://firefox-ci-tc.services.mozilla.com/tasks/ARx6P9gKSJOtiT20GrPKjQ/runs/1/logs/public/logs/live_backing.log)

Reply:
Oh, and if you meant specifically what the `verb` thing is about, it's just improving the log message to make it more clear what treescript is doing.